src/transformers/pipelines/image_classification.py

```diff
@@ -179,7 +180,7 @@ def postprocess(self, model_outputs, function_to_apply=None, top_k=5):
         if top_k > self.model.config.num_labels:
             top_k = self.model.config.num_labels
 
-        outputs = model_outputs["logits"][0]
+        outputs = model_outputs["logits"][0].to(torch.float32)
         outputs = outputs.numpy()
```

Comment:
We can't do this directly, as the outputs might not be torch tensors. 

We'll need to do something like: 
```suggestion
        outputs = model_outputs["logits"][0]
        
        if self.framework == "pt":
            # If torch_dtype is bf16 or fp16
            outputs = outputs.float().numpy()
        else:
            outputs = outputs.numpy()
```